src/service/UserService.php

```diff
@@ -0,0 +1,190 @@
+     * 
+     * @return array
+     */
+    public function connect(
+        string $password, 
+        ConnectionModel $connectionModel
+    ) :array {
+        $dbPassword = password_verify($password, $connectionModel->password);
+
+        if ($dbPassword) {
+            $_SESSION["connected"] = true;
+            $_SESSION["button"] = "disconnect";
+            $_SESSION["user"]= [
+                "first_name" => $connectionModel->firstName,
+                "role" => $connectionModel->role,
+                "is_allowed" => $connectionModel->isAllowed
+            ];
+
+            $template = "home.html.twig";
+            $data = [
+                "message" => 
+                "Bonjour " . $connectionModel->firstName . " vous êtes connecté.",
+            ];
+
+        } else {
+            $template = "login.html.twig";
+            $data = [
+                "error" => "Problème d'identification."
+            ];
+        }
+        $result = [
+            "template" => $template,
+            "data" => $data
+        ];
+
+        return $result;
+    }
+
+    /**
+     * Summary of getUserEntity
+     * 
+     * @param string $username come from the connection form
+     * @param string $password come from the connection form
+     * 
+     * @return \App\entity\UserEntity
+     */
+    public function getUserEntity(
+        string $username, string $password
+    ) :UserEntity {
+        $userRepository = new UserRepository;
+
+        $result = $userRepository->getUser($username);
+
+        if ($result !== []) {
+            $creationDate = $result[0]["creation_date"];
+            $creationDate = DateTime::createFromFormat(
+                "Y-m-d H:i:s", 
+                date("Y-m-d H:i:s")
+            );
+
+            $updateDate = $result[0]["last_update_date"];
+            $updateDate = DateTime::createFromFormat(
+                "Y-m-d H:i:s", 
+                date("Y-m-d H:i:s")
+            );
+
+            $allowed = boolval($result[0]["is_allowed"]);
+
+            $user = new UserEntity(
+                $result[0]["id"], 
+                $result[0]["name"], 
+                $result[0]["first_name"], 
+                $result[0]["username"], 
+                $result[0]["email"], 
+                $result[0]["password"], 
+                $result[0]["role"], 
+                $creationDate, 
+                $updateDate,
+                $allowed
+            );
+        } else {
+            $user = null;
+        }
+        return $user;
+    }
+
+    /**
+     * Summary of checkData
+     * verify the data entered by the user and verify if they are empty
+     * 
+     * @param string $username come from the connection form
+     * @param string $password come from the connection form
+     * 
+     * @return bool
+     */
+    public function checkData(string $username, string $password)
+    {
+        if ($username === "" || $password === "") {
```

Comment:
tu peux faire un early return 
```suggestion
        if ($username === "" || $password === "") return false;
        return true; // pas besoin de else ici.
```

Reply:
corrigé

Reply:
En fait je rechange parce que la syntaxe en une seule ligne est mise en warning par codesniffer